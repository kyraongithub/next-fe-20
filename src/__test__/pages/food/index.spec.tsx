import FoodPage from '@/pages/food';
import { render } from '@testing-library/react';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe('Food Page', () => {
  beforeEach(async () => {
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');

    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    }));
  });
  it('should render food page', () => {
    const { container } = render(<FoodPage />);
    expect(container).toMatchSnapshot();
  });
});
