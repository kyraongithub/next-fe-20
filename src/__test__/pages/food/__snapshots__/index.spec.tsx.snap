// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Page should render food page 1`] = `
<div>
  <div>
    Food list page
    <p>
      loading
    </p>
  </div>
</div>
`;
